hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -89,7 +88,7 @@ protected ClientInvocation(HazelcastClientInstanceImpl client,
         retryTimeoutPointInMillis = System.currentTimeMillis() + waitTimeResolved;
 
         clientInvocationFuture = new ClientInvocationFuture(this, client, clientMessage);
-
+        logger = client.getClientLoggingService().getLogger(ClientInvocation.class);
```

Comment:
You don't want to do this lookup for every invocation. I would move this method to the invocation-service and if a logger is needed here do:

```
logger = invocationService.invocationLogger
```

Perhaps less pretty, but no unneeded work is added at the fast path.